scripts/broadcast-nmdp-discover.nse

```diff
@@ -0,0 +1,231 @@
+        local h = math.floor(uptime_num / 3600)
+        local m = math.floor((uptime_num - (h * 3600)) / 60)
+        uptime = h .. "h" .. m .. "m"
+
+      -- Software ID
+      elseif tlv_type == 0x0b then
+        software_id = tlv_value
+
+      -- Board
+      elseif tlv_type == 0x0c then
+        board = tlv_value
+
+      -- Unpacking - unsigned integer
+      elseif tlv_type == 0x0e then
+        if tlv_value:byte(1) == 0x01 then
+          unpacking = "None"
+        else
+          unpacking = "Unknown"
+        end
+
+      -- Interface
+      elseif tlv_type == 0x10 then
+        interface = tlv_value
+
+      -- Unknown
+      else
+        stdnse.print_debug(2, "Unknown TLV Type: %s", tlv_type)
+      end
+    end
+
+    local str = ("MAC Address: %s, IP Address: %s; Identity: %s; Version: %s; Platform: %s; Software ID: %s; Uptime: %s; Board: %s; Unpacking: %s; Interface: %s"):format(
+      mac_address, p.ip_src, identity, version, platform, software_id, uptime, board, unpacking, interface)
+
+    table.insert(responses, str)
+
+    ::continue::
+  end
+
+  condvar("signal")
+end
+
+-- Returns the network interface used to send packets to a target host.
+--@param target host to which the interface is used.
+--@return interface Network interface used for target host.
+local getInterface = function(target)
+  -- First, create dummy UDP connection to get interface
+  local sock = nmap.new_socket()
+  local status, err = sock:connect(target, "12345", "udp")
+  if not status then
+    stdnse.verbose1("%s", err)
+    return
+  end
+  local status, address, _, _, _ = sock:get_info()
+  if not status then
+    stdnse.verbose1("%s", err)
+    return
+  end
+  for _, interface in pairs(nmap.list_interfaces()) do
+    if interface.address == address then
+      return interface
+    end
+  end
+end
+
+action = function()
+  local host = { ip = arg_address or "255.255.255.255" }
+
+  -- Check if a valid interface was provided
+  local interface = nmap.get_interface()
+  if interface then
+    interface = nmap.get_interface_info(interface)
+  else
+    interface = getInterface(host)
+  end
+  if not interface then
+    return ("\n ERROR: Couldn't get interface for %s"):format(host)
+  end
+
+  -- Launch listener thread
+  local results = {}
+  local timeout = (tonumber(arg_timeout) or 5) * 1000
+  stdnse.new_thread(nmdpListen, interface, timeout, results)
+  stdnse.sleep(0.5)
+
+  -- send two packets, just in case
+  local port = { number = 5678, protocol = "udp" }
+  local socket = nmap.new_socket("udp")
+  socket:set_timeout(500)
+  for i=1,2 do
+    local status = socket:sendto(host, port, "\x00\x00\x00\x00")
+    if ( not(status) ) then
+      return stdnse.format_output(false, "Failed to send broadcast probe")
+    end
+  end
+
+  -- Wait for listener thread to finish
+  local condvar = nmap.condvar(results)
+  condvar("wait")
+
+  -- Create output table
+  local output = stdnse.output_table()
```

Comment:
In this case, there's not a reason to use a `stdnse.output_table()` since the responses aren't being put in any particular order. You could instead sort them just before returning to ensure repeatable ordering.

Reply:
How do I sort the `results` table (and remove duplicates) without using `stdnse.output_table()` ? Keeping in mind that duplicates should only be removed if every element in one row is identical to another (ie, duplicate MAC addresses and IP addresses are expected).